src/bijectors/corr.jl

```diff
@@ -0,0 +1,109 @@
+# See stan doc for parametrization method:
+# https://mc-stan.org/docs/2_23/reference-manual/correlation-matrix-transform-section.html
+# (7/30/2020) their "manageable expression" is wrong...
+
+struct CorrBijector <: Bijector{2} end
+
+(b::CorrBijector)(X::AbstractMatrix{<:Real}) = link_lkj(X)
+(b::CorrBijector)(X::AbstractArray{<:AbstractMatrix{<:Real}}) = map(b, X)
+
+(ib::Inverse{<:CorrBijector})(Y::AbstractMatrix{<:Real}) = inv_link_lkj(Y)
+(ib::Inverse{<:CorrBijector})(Y::AbstractArray{<:AbstractMatrix{<:Real}}) = map(ib, Y)
+
+
+logabsdetjac(::Inverse{CorrBijector}, y::AbstractMatrix{<:Real}) = log_abs_det_jac_lkj(y)
+function logabsdetjac(b::CorrBijector, X::AbstractMatrix{<:Real})
+    
+    return -log_abs_det_jac_lkj(b(X)) # It may be more efficient if we can use un-contraint value to prevent call of b
+end
+logabsdetjac(b::CorrBijector, X::AbstractArray{<:AbstractMatrix{<:Real}}) = mapvcat(X) do x
+    logabsdetjac(b, x)
+end
+
+
+function log_abs_det_jac_lkj(y)
+    # it's defined on inverse mapping
+    K = size(y, 1)
+    
+    z = tanh.(y)
+    left = eltype(y)(0)
+    for i = 1:(K-1), j = (i+1):K
+        left += (K-i-1) * log(1 - z[i, j]^2)
+    end
+    
+    right = eltype(y)(0)
+    for i = 1:(K-1), j = (i+1):K
+        right += log(cosh(y[i, j])^2)
+    end
+    
+    return left / 2 - right
+end
+
+function inv_link_w_lkj(y)
```

Comment:
I guess this should just be
```suggestion
function (ib::Inverse{<:CorrBijector})(y::AbstractMatrix{<:Real})
```
and return `w' * w` instead. All these different functions with (IMO) not really descriptive names should be avoided if possible.

Reply:
I'm guessing the reason why @yiyuezhuo has done this is to be able to define custom adjoints. Not possible to do so if we use `(b::Bijector)(x)`. 

So I think leaving the `(inv_)link_w_lkj` is fine, but redefine `(inv_)link_lkj` to their corresponding `b::Bijector` definitions.

Reply:
For Zygote (or rather ChainRules nowadays) it's definitely possible to define adjoints for `(::Bijector)(...)` (see e.g. the support of Distances in Zygote). Does it not work for other AD backends?

Reply:
I defined custom adjoint on `link_w_lkj` but not on `link_lkj`, so link_w_lkj is just an "anonymous" code block requiring a custom gradient.

Reply:
Hmm I'm not sure how that relates to my comment? I just wanted to point out that in Zygote and ChainRules you can also define adjoints for functions such as `(ib::Inverse{<:CorrBijector})(y::AbstractMatrix{<:Real}) = ...` if needed - there is no need to introduce some function `inv_link_lkj` for this purpose.

Reply:
Woah, really? If so I guess there's no need. Back when we started this overhaul of Bijectors, it was not possible to define adjoints for those kind of signatures. If that's not the case anymore, then I agree with your comments @devmotion

Reply:
In Zygote you can write something like https://github.com/FluxML/Zygote.jl/blob/956575ee2c732dee25324b59ba43fbb471a52d9a/src/lib/distances.jl#L3-L10. However, we should definitely use ChainRules (https://www.juliadiff.org/ChainRulesCore.jl/dev/index.html#frule-and-rrule) which supports this as well.

Reply:
I mean, let's say these is a function `f` which needs AD:

```julia
function f(x)
  y = f1(x)
  z = f2(y)
  return z
end
```
where `f1` is AD-friendly and `f2` needs a custom adjoint. So I wrote an adjoint for `f2` only. Are you suggesting to write an adjoint for `f` and call `pullback` for `f1` there? It's not clear to me.